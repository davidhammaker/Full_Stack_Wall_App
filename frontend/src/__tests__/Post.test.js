import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Post from '../components/post';


configure({ adapter: new Adapter() });


it('displays a post', () => {
  const testPost = {
    id: 1,
    content: 'test 1',
    creator: 'testUser',
    date_posted: '2019-08-14T00:17:28.427013Z'
  };
  const post = shallow(
    <Post key={ testPost.id } post={ testPost } />
  );

  expect(post.text()).toContain('test 1');
});

it('displays a re-formatted date', () => {
  const testPost = {
    id: 1,
    content: 'test 1',
    creator: 'testUser',
    date_posted: '2019-08-14T00:17:28.427013Z'
  };
  const post = shallow(
    <Post key={ testPost.id } post={ testPost } />
  );

  expect(post.text()).toContain('12:17 AM, 08/14/2019');
});

it('displays post correctly', () => {
  const testPost = {
    id: 1,
    content: 'test 1',
    creator: 'testUser',
    date_posted: '2019-08-14T00:17:28.427013Z'
  };
  const post = renderer.create(
    <Post key={ testPost.id } post={ testPost } />
  );
  let tree = post.toJSON();
  expect(tree).toMatchSnapshot();
});
