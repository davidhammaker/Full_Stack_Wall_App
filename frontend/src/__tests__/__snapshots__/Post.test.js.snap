// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays post correctly 1`] = `
<div
  className="p-2"
>
  <h5>
    testUser
    <small
      className="float-right font-italic small-font"
    >
      12:17 AM, 08/14/2019
    </small>
  </h5>
  <p
    className="pl-2 large-font"
  >
    test 1
  </p>
</div>
`;
